backend/src/schema/resolvers/passwordReset.js

```diff
@@ -2,39 +2,47 @@ import uuid from 'uuid/v4'
 import bcrypt from 'bcryptjs'
 
 export async function createPasswordReset(options) {
-  const { driver, code, email, issuedAt = new Date() } = options
+  const { driver, nonce, email, issuedAt = new Date() } = options
   const session = driver.session()
-  const cypher = `
+  let response = {}
+  try {
+    const cypher = `
       MATCH (u:User)-[:PRIMARY_EMAIL]->(e:EmailAddress {email:$email})
-      CREATE(pr:PasswordReset {code: $code, issuedAt: datetime($issuedAt), usedAt: NULL})
+      CREATE(pr:PasswordReset {nonce: $nonce, issuedAt: datetime($issuedAt), usedAt: NULL})
       MERGE (u)-[:REQUESTED]->(pr)
-      RETURN u
+      RETURN e, pr, u
       `
-  const transactionRes = await session.run(cypher, {
-    issuedAt: issuedAt.toISOString(),
-    code,
-    email,
-  })
-  const users = transactionRes.records.map(record => record.get('u'))
-  session.close()
-  return users
+    const transactionRes = await session.run(cypher, {
+      issuedAt: issuedAt.toISOString(),
+      nonce,
+      email,
+    })
+    const records = transactionRes.records.map(record => {
+      const { email } = record.get('e').properties
+      const { nonce } = record.get('pr').properties
+      const { name } = record.get('u').properties
+      return { email, nonce, name }
+    })
+    response = records[0] || {}
+  } finally {
+    session.close()
+  }
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-10T07:59:13Z" title="Tuesday, September 10th 2019, 9:59:13 am +02:00">Sep 10, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -2,39 +2,47 @@ import uuid from 'uuid/v4'
 import bcrypt from 'bcryptjs'
 
 export async function createPasswordReset(options) {
-  const { driver, code, email, issuedAt = new Date() } = options
+  const { driver, nonce, email, issuedAt = new Date() } = options
   const session = driver.session()
-  const cypher = `
+  let response = {}
+  try {
+    const cypher = `
       MATCH (u:User)-[:PRIMARY_EMAIL]->(e:EmailAddress {email:$email})
-      CREATE(pr:PasswordReset {code: $code, issuedAt: datetime($issuedAt), usedAt: NULL})
+      CREATE(pr:PasswordReset {nonce: $nonce, issuedAt: datetime($issuedAt), usedAt: NULL})
       MERGE (u)-[:REQUESTED]->(pr)
-      RETURN u
+      RETURN e, pr, u
       `
-  const transactionRes = await session.run(cypher, {
-    issuedAt: issuedAt.toISOString(),
-    code,
-    email,
-  })
-  const users = transactionRes.records.map(record => record.get('u'))
-  session.close()
-  return users
+    const transactionRes = await session.run(cypher, {
+      issuedAt: issuedAt.toISOString(),
+      nonce,
+      email,
+    })
+    const records = transactionRes.records.map(record => {
+      const { email } = record.get('e').properties
+      const { nonce } = record.get('pr').properties
+      const { name } = record.get('u').properties
+      return { email, nonce, name }
+    })
+    response = records[0] || {}
+  } finally {
+    session.close()
+  }
```


what happens if there is an error here? 
will it return some long difficult to understand error message?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-10T14:14:56Z" title="Tuesday, September 10th 2019, 4:14:56 pm +02:00">Sep 10, 2019</time>_

---


I guess so, yes